core/src/util/vboMesh.cpp

```diff
@@ -68,15 +68,22 @@ void VboMesh::addVertex(GLbyte* _vertex) {
 void VboMesh::addVertices(GLbyte* _vertices, int _nVertices) {
 
     if (m_isUploaded) {
-        logMsg("%s\n", "VboMesh cannot add vertices after upload!");
+        logMsg("ERROR: VboMesh cannot add vertices after upload!\n");
         return;
     }
     
     // Only add up to 65535 vertices, any more will overflow our 16-bit indices
     int indexSpace = MAX_INDEX_VALUE - m_nVertices;
-    if (_nVertices > indexSpace) {
-        _nVertices = indexSpace;
-        logMsg("WARNING: Tried to add more vertices than available in index space\n");
+    if (_nVertices > indexSpace || m_backupMesh) {
+        if (!m_backupMesh) {
+            m_backupMesh.reset(new VboMesh(m_vertexLayout, m_drawMode));
+        }
+        if (_nVertices > MAX_INDEX_VALUE) {
```

Comment:
This check should happen before backupMesh is created. @blair1618 is looking into this.